src/Application.vala

```diff
@@ -103,12 +103,52 @@ public class Music.Application : Gtk.Application {
         settings.bind ("window-maximized", main_window, "maximized", SettingsBindFlags.SET);
     }
 
+    File[] listDirectory(string directory) {
+        var dir = Dir.open(directory, 0);
+
+        string? name = null;
+        File[] elements = {};
+
+        while((name = dir.read_name ()) != null) {
+            string filePath = Path.build_filename (directory, name);
+            File file = File.new_for_path(filePath);
+
+            elements += file;
+        }
+
+        return elements;
+    }
+
+    private bool isDirectory(string file) {
+        return FileUtils.test(file, FileTest.IS_DIR);
+    }
+
+    private void loopThroughFiles(File[] files) {
+        foreach (var file in files) {
+            var filePath = file.get_path();
+            var isDirectory = isDirectory(filePath);
+
+            if (!isDirectory) {
```

Comment:
Can you please reverse this if? Like check if file is directory and then handle it as a directory rather than checking if it's a file